php/authentication.php

```diff
@@ -198,5 +202,42 @@ function RedirectToHttpsIfRequired($configData){
     }
 	StopTimer("RedirectToHttpsIfRequired");
 }
+function VerifyPassword($user, $password) {
+	global $configData;
+	switch ($user->AuthVersion) {
+		case 1:
+			$correctPasswordHash = $user->PasswordHash;
+			$userSalt = $user->Salt;
+			$passwordIterations = intval($user->PasswordIterations);
+			$passwordHash = HashPassword($password, $userSalt, $passwordIterations, $configData);
+			if($correctPasswordHash != $passwordHash)
+				return "INCORRECT_PASSWORD";
+			UpdateUserPassword($user->Id, $password, true);
+			break;
+		case 2:
+			if(!password_verify($password, $user->PasswordHash))
+				return "INCORRECT_PASSWORD";
+			break;
+		default:
+			return "INVALID_AUTH_VERSION";
+	}
+	return "SUCCESS";
+}
 
+function UpdateUserPassword($userId, $plainTextPassword, $update_loadedUserData) {
+	global $loggedInUser, $userData, $userDbInterface;
+	
+	// bcrypt generates new salt, number of iterations and hashed password and stores them by itself.
+	$newPasswordHash = password_hash($plainTextPassword, PASSWORD_BCRYPT);
+	if($update_loadedUserData) {
+		// the below part should not run if it's an admin changing the password of another account.
+		$userData->UserModels[$userId]->Salt = OVERRIDE_UNUSED; // doesn't matter with bcrypt
+		$userData->UserModels[$userId]->PasswordIterations = OVERRIDE_UNUSED; // doesn't matter with bcrypt
+	
+		$userData->UserModels[$userId]->PasswordHash = $newPasswordHash; // only important field with bcrypt
+		$userData->UserModels[$userId]->AuthVersion = AUTH_BCRYPT; // tell site we're using AUTH_BCRYPT now.
+	}
```

Comment:
Since this function is used with the register method, not having the block makes it tell the user that registration failed because they provided an incorrect password (fails to login while registering), tho i did not know about `isset(...)` and that looks like exactly what i wanted here, so I'll use that here.